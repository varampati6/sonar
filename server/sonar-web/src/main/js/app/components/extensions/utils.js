/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
// @flow
import { getExtensionFromCache } from '../../utils/installExtensionsHandler';

const installScript = (key: string) => {
  return new Promise(resolve => {
    const scriptTag = document.createElement('script');
    scriptTag.src = `${window.baseUrl}/static/${key}.js`;
    scriptTag.onload = resolve;
    document.getElementsByTagName('body')[0].appendChild(scriptTag);
  });
};

export const getExtensionStart = (key: string) =>
  new Promise((resolve, reject) => {
    const fromCache = getExtensionFromCache(key);
    if (fromCache) {
      return resolve(fromCache);
    }

    installScript(key).then(() => {
      const start = getExtensionFromCache(key);
      if (start) {
        resolve(start);
      } else {
        reject();
      }
    });
  });
