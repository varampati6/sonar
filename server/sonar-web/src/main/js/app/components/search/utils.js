/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
// @flow
import { sortBy } from 'lodash';

const ORDER = ['DEV', 'VW', 'SVW', 'TRK', 'BRC', 'FIL', 'UTS'];

export function sortQualifiers(qualifiers: Array<string>) {
  return sortBy(qualifiers, qualifier => ORDER.indexOf(qualifier));
}

export type Component = {
  isFavorite?: boolean,
  isRecentlyBrowsed?: boolean,
  key: string,
  match?: string,
  name: string,
  organization?: string,
  project?: string,
  qualifier: string
};

export type Results = { [qualifier: string]: Array<Component> };

export type More = { [string]: number };
