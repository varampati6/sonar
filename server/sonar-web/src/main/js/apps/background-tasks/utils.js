/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
/* @flow */
import { STATUSES, ALL_TYPES, CURRENTS } from './constants';
import { Task } from './types';

export function updateTask(tasks: Task[], newTask: Task) {
  return tasks.map(task => (task.id === newTask.id ? newTask : task));
}

export function mapFiltersToParameters(filters: Object = {}) {
  const parameters = {};

  if (filters.status === STATUSES.ALL) {
    parameters.status = [
      STATUSES.PENDING,
      STATUSES.IN_PROGRESS,
      STATUSES.SUCCESS,
      STATUSES.FAILED,
      STATUSES.CANCELED
    ].join();
  } else if (filters.status === STATUSES.ALL_EXCEPT_PENDING) {
    parameters.status = [
      STATUSES.IN_PROGRESS,
      STATUSES.SUCCESS,
      STATUSES.FAILED,
      STATUSES.CANCELED
    ].join();
  } else {
    parameters.status = filters.status;
  }

  if (filters.taskType !== ALL_TYPES) {
    parameters.type = filters.taskType;
  }

  if (filters.currents !== CURRENTS.ALL) {
    parameters.onlyCurrents = true;
  }

  if (filters.minSubmittedAt) {
    parameters.minSubmittedAt = filters.minSubmittedAt;
  }

  if (filters.maxExecutedAt) {
    parameters.maxExecutedAt = filters.maxExecutedAt;
  }

  if (filters.query) {
    parameters.componentQuery = filters.query;
  }

  if (filters.lastPage !== 1) {
    parameters.p = filters.lastPage;
  }

  return parameters;
}

const ONE_SECOND = 1000;
const ONE_MINUTE = 60 * ONE_SECOND;

function format(int, suffix) {
  return `${int}${suffix}`;
}

export function formatDuration(value: ?number) {
  if (!value) {
    return '';
  }
  if (value >= ONE_MINUTE) {
    const minutes = Math.round(value / ONE_MINUTE);
    return format(minutes, 'min');
  } else if (value >= ONE_SECOND) {
    const seconds = Math.round(value / ONE_SECOND);
    return format(seconds, 's');
  } else {
    return format(value, 'ms');
  }
}
