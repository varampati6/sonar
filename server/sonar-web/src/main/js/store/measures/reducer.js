/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import {
  RECEIVE_COMPONENT_MEASURE,
  RECEIVE_COMPONENT_MEASURES,
  RECEIVE_COMPONENTS_MEASURES
} from './actions';

const byMetricKey = (state = {}, action = {}) => {
  if (action.type === RECEIVE_COMPONENT_MEASURE) {
    return { ...state, [action.metricKey]: action.value };
  }

  if (action.type === RECEIVE_COMPONENT_MEASURES) {
    return { ...state, ...action.measures };
  }

  return state;
};

const reducer = (state = {}, action = {}) => {
  if ([RECEIVE_COMPONENT_MEASURE, RECEIVE_COMPONENT_MEASURES].includes(action.type)) {
    const component = state[action.componentKey];
    return { ...state, [action.componentKey]: byMetricKey(component, action) };
  }

  if (action.type === RECEIVE_COMPONENTS_MEASURES) {
    const newState = { ...state };
    Object.keys(action.componentsWithMeasures).forEach(componentKey => {
      Object.assign(newState, {
        [componentKey]: byMetricKey(state[componentKey], {
          type: RECEIVE_COMPONENT_MEASURES,
          measures: action.componentsWithMeasures[componentKey]
        })
      });
    });
    return newState;
  }

  return state;
};

export default reducer;

export const getComponentMeasure = (state, componentKey, metricKey) => {
  const component = state[componentKey];
  return component && component[metricKey];
};

export const getComponentMeasures = (state, componentKey) => state[componentKey];
