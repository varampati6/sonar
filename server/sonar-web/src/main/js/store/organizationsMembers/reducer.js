/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { uniq } from 'lodash';
import { actions } from './actions';

export const getOrganizationMembersLogins = (state, organization) => {
  if (organization && state[organization]) {
    return state[organization].members || [];
  }
  return [];
};

export const getOrganizationMembersState = (state, organization) =>
  (organization && state[organization] ? state[organization] : {});

const organizationMembers = (state = {}, action = {}) => {
  const members = state.members || [];
  switch (action.type) {
    case actions.UPDATE_STATE:
      return { ...state, ...action.stateChanges };
    case actions.RECEIVE_MEMBERS:
      return {
        ...state,
        ...action.stateChanges,
        members: action.members.map(member => member.login)
      };
    case actions.RECEIVE_MORE_MEMBERS:
      return {
        ...state,
        ...action.stateChanges,
        members: uniq(members.concat(action.members.map(member => member.login)))
      };
    case actions.ADD_MEMBER: {
      const withNew = [...members, action.member.login].sort();
      return {
        ...state,
        total: state.total + 1,
        members: withNew
      };
    }
    case actions.REMOVE_MEMBER: {
      const withoutDeleted = state.members.filter(login => login !== action.member.login);
      if (withoutDeleted.length === state.members.length) {
        return state;
      }
      return {
        ...state,
        total: state.total - 1,
        members: withoutDeleted
      };
    }
    default:
      return state;
  }
};

const organizationsMembers = (state = {}, action = {}) => {
  const organization = state[action.organization] || {};
  switch (action.type) {
    case actions.UPDATE_STATE:
    case actions.RECEIVE_MEMBERS:
    case actions.RECEIVE_MORE_MEMBERS:
    case actions.ADD_MEMBER:
    case actions.REMOVE_MEMBER:
      return {
        ...state,
        [action.organization]: organizationMembers(organization, action)
      };
    default:
      return state;
  }
};

export default organizationsMembers;
