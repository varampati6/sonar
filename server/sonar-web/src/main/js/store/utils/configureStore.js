/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const composed = [];

if (process.env.NODE_ENV !== 'production') {
  const createLogger = require('redux-logger');
  middlewares.push(createLogger());

  composed.push(window.devToolsExtension ? window.devToolsExtension() : f => f);
}

const finalCreateStore = compose(applyMiddleware(...middlewares), ...composed)(createStore);

export default function configureStore(rootReducer, initialState) {
  return finalCreateStore(rootReducer, initialState);
}

export const configureTestStore = (rootReducer, initialState) =>
  createStore(rootReducer, initialState);
