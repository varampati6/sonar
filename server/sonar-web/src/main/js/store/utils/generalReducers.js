/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
// Author: Christoffer Niska <christofferniska@gmail.com>
// https://gist.github.com/crisu83/42ecffccad9d04c74605fbc75c9dc9d1
import { uniq } from 'lodash';

/**
 * Creates a reducer that manages a single value.
 *
 * @param {function(state, action)} shouldUpdate
 * @param {function(state, action)} shouldReset
 * @param {function(state, action)} getValue
 * @param {*} defaultValue
 * @returns {function(state, action)}
 */
export const createValue = (
  shouldUpdate = () => true,
  shouldReset = () => false,
  getValue = (state, action) => action.payload,
  defaultValue = null
) => (state = defaultValue, action = {}) => {
  if (shouldReset(state, action)) {
    return defaultValue;
  }
  if (shouldUpdate(state, action)) {
    return getValue(state, action);
  }
  return state;
};

/**
 * Creates a reducer that manages a map.
 *
 * @param {function(state, action)} shouldUpdate
 * @param {function(state, action)} shouldReset
 * @param {function(state, action)} getValues
 * @returns {function(state, action)}
 */
export const createMap = (
  shouldUpdate = () => true,
  shouldReset = () => false,
  getValues = (state, action) => action.payload
) =>
  createValue(
    shouldUpdate,
    shouldReset,
    (state, action) => ({ ...state, ...getValues(state, action) }),
    {}
  );

/**
 * Creates a reducer that manages a set.
 *
 * @param {function(state, action)} shouldUpdate
 * @param {function(state, action)} shouldReset
 * @param {function(state, action)} getValues
 * @returns {function(state, action)}
 */
export const createSet = (
  shouldUpdate = () => true,
  shouldReset = () => false,
  getValues = (state, action) => action.payload
) =>
  createValue(
    shouldUpdate,
    shouldReset,
    (state, action) => uniq([...state, ...getValues(state, action)]),
    []
  );

/**
 * Creates a reducer that manages a flag.
 *
 * @param {function(state, action)} shouldTurnOn
 * @param {function(state, action)} shouldTurnOff
 * @param {bool} defaultValue
 * @returns {function(state, action)}
 */
export const createFlag = (shouldTurnOn, shouldTurnOff, defaultValue = false) => (
  state = defaultValue,
  action = {}
) => {
  if (shouldTurnOn(state, action)) {
    return true;
  }
  if (shouldTurnOff(state, action)) {
    return false;
  }
  return state;
};
