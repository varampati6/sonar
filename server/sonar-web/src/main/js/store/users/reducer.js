/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { combineReducers } from 'redux';
import { uniq, keyBy } from 'lodash';
import { RECEIVE_CURRENT_USER, RECEIVE_USER } from './actions';
import { actions as membersActions } from '../organizationsMembers/actions';

const usersByLogin = (state = {}, action = {}) => {
  switch (action.type) {
    case RECEIVE_CURRENT_USER:
    case RECEIVE_USER:
      return { ...state, [action.user.login]: action.user };
    case membersActions.RECEIVE_MEMBERS:
    case membersActions.RECEIVE_MORE_MEMBERS:
      return { ...state, ...keyBy(action.members, 'login') };
    case membersActions.ADD_MEMBER:
      return { ...state, [action.member.login]: action.member };
    default:
      return state;
  }
};

const userLogins = (state = [], action = {}) => {
  switch (action.type) {
    case RECEIVE_CURRENT_USER:
    case RECEIVE_USER:
      return uniq([...state, action.user.login]);
    case membersActions.RECEIVE_MEMBERS:
    case membersActions.RECEIVE_MORE_MEMBERS:
      return uniq([...state, action.members.map(member => member.login)]);
    case membersActions.ADD_MEMBER: {
      return uniq([...state, action.member.login]).sort();
    }
    default:
      return state;
  }
};

const currentUser = (state = null, action = {}) => {
  if (action.type === RECEIVE_CURRENT_USER) {
    return action.user;
  }
  return state;
};

export default combineReducers({ usersByLogin, userLogins, currentUser });

export const getCurrentUser = state => state.currentUser;
export const getUserLogins = state => state.userLogins;
export const getUserByLogin = (state, login) => state.usersByLogin[login];
export const getUsersByLogins = (state, logins) =>
  logins.map(login => getUserByLogin(state, login));
export const getUsers = state => getUsersByLogins(state, getUserLogins(state));
