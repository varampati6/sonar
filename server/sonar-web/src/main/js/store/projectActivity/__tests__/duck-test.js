/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { configureTestStore } from '../../utils/configureStore';
import reducer, {
  receiveProjectActivity,
  getAnalyses,
  getPaging,
  addEvent,
  changeEvent,
  deleteEvent,
  deleteAnalysis
} from '../duck';

const PROJECT = 'project-foo';

const ANALYSES = [
  {
    key: 'AVgGkRvCrrTJiPpCD-rG',
    date: '2016-10-27T16:33:50+0200',
    events: [
      {
        key: 'AVjUDBiSiXOcXjpycvde',
        category: 'VERSION',
        name: '2.18-SNAPSHOT'
      }
    ]
  },
  {
    key: 'AVgFqeOSKpGuA48ADATE',
    date: '2016-10-27T12:21:15+0200',
    events: []
  },
  {
    key: 'AVgAgC1Vdo07z3PUnnkt',
    date: '2016-10-26T12:17:29+0200',
    events: [
      {
        key: 'AVkWNYNYr4pSN7TrXcjY',
        category: 'OTHER',
        name: 'foo'
      }
    ]
  }
];

const PAGING = {
  total: 3,
  pageIndex: 1,
  pageSize: 100
};

describe('actions', () => {
  it('addEvent', () => {
    expect(addEvent('foo', { key: 'bar' })).toMatchSnapshot();
  });

  it('changeEvent', () => {
    expect(changeEvent('foo', { name: 'bar' })).toMatchSnapshot();
  });

  it('deleteEvent', () => {
    expect(deleteEvent('foo', 'bar')).toMatchSnapshot();
  });

  it('deleteAnalysis', () => {
    expect(deleteAnalysis('foo', 'bar')).toMatchSnapshot();
  });
});

describe('selectors', () => {
  it('getAnalyses', () => {
    const store = configureTestStore(reducer);
    store.dispatch(receiveProjectActivity(PROJECT, ANALYSES, PAGING));
    expect(getAnalyses(store.getState(), PROJECT)).toMatchSnapshot();
  });

  it('getPaging', () => {
    const store = configureTestStore(reducer);
    store.dispatch(receiveProjectActivity(PROJECT, ANALYSES, PAGING));
    expect(getPaging(store.getState(), PROJECT)).toMatchSnapshot();
  });
});
