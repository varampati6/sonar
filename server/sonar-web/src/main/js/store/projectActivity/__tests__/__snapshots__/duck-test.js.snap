// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions addEvent 1`] = `
Object {
  "analysis": "foo",
  "event": Object {
    "key": "bar",
  },
  "type": "ADD_PROJECT_ACTIVITY_EVENT",
}
`;

exports[`actions changeEvent 1`] = `
Object {
  "changes": Object {
    "name": "bar",
  },
  "event": "foo",
  "type": "CHANGE_PROJECT_ACTIVITY_EVENT",
}
`;

exports[`actions deleteAnalysis 1`] = `
Object {
  "analysis": "bar",
  "project": "foo",
  "type": "DELETE_PROJECT_ACTIVITY_ANALYSIS",
}
`;

exports[`actions deleteEvent 1`] = `
Object {
  "analysis": "foo",
  "event": "bar",
  "type": "DELETE_PROJECT_ACTIVITY_EVENT",
}
`;

exports[`selectors getAnalyses 1`] = `
Array [
  Object {
    "date": "2016-10-27T16:33:50+0200",
    "events": Array [
      Object {
        "category": "VERSION",
        "key": "AVjUDBiSiXOcXjpycvde",
        "name": "2.18-SNAPSHOT",
      },
    ],
    "key": "AVgGkRvCrrTJiPpCD-rG",
  },
  Object {
    "date": "2016-10-27T12:21:15+0200",
    "events": Array [],
    "key": "AVgFqeOSKpGuA48ADATE",
  },
  Object {
    "date": "2016-10-26T12:17:29+0200",
    "events": Array [
      Object {
        "category": "OTHER",
        "key": "AVkWNYNYr4pSN7TrXcjY",
        "name": "foo",
      },
    ],
    "key": "AVgAgC1Vdo07z3PUnnkt",
  },
]
`;

exports[`selectors getPaging 1`] = `
Object {
  "pageIndex": 1,
  "pageSize": 100,
  "total": 3,
}
`;
